plugin/builtin/configmapgenerator/ConfigMapGenerator.go

```diff
@@ -16,6 +16,10 @@ type plugin struct {
 	rf  *resmap.Factory
 	types.GeneratorOptions
 	types.ConfigMapArgs
+	Metadata struct {
+		Name      string `json:"name,omitempty" yaml:"name,omitempty"`
+		Namespace string `json:"namespace,omitempty" yaml:"namespace,omitempty"`
+	} `json:"metadata,omitempty" yaml:"metadata,omitempty"`
```

Comment:
can this reuse types.ObjectMeta from  #1284 (once it's in)?

Reply:
Yup, was thinking about that but wanted to see what you think before making bigger changes.

Reply:
so if i wrote the comment for this PR it would be something like

```
Since plugin config files are k8s objects, they must have a
metadata name field and optionally namespace.  We
might as well use those fields as _data for the plugin_
in situations where the plugin wants a field called `name`
and/or `namespace`, en lieu of declaring a new field
called `name` (or `namespace`).
```

is that the idea? seems fine.

Reply:
yeah, sorry, should have explained better.

Reply:
can you paraphrase that or something and throw it in the description?